packages/affiliates/apps/MultiDappMiningRewards.js

```diff
@@ -0,0 +1,44 @@
+const lodash = require("lodash");
+const Promise = require("bluebird");
+
+const { getWeb3 } = require("@uma/common");
+const { getAbi } = require("@uma/core");
+const { BigQuery } = require("@google-cloud/bigquery");
+
+const { DappMining } = require("../libs/affiliates");
+const Queries = require("../libs/bigquery");
+const {
+  generateDappMiningConfig,
+  dappMiningTemplate,
+  createGithubIssue,
+  saveToDisk,
+  makeDappMiningFilename,
+  makeUnixPipe
+} = require("../libs/affiliates/utils");
+
+// This runs the full pipeline for dapp mining all emps in the configuration.
+// This will probably be deprecated in favor of a more composable approach.
+// Currently theres no good "whitelist" available for dapp mining emps, like there is for dev mining.
+const App = env => async params => {
+  const web3 = getWeb3();
+
+  const empAbi = getAbi("ExpiringMultiParty");
+  const client = new BigQuery();
+  const queries = Queries({ client });
+
+  const dappmining = DappMining({ empAbi, queries, web3 });
+  params = lodash.castArray(params);
+  await Promise.map(params, async param => {
```

Comment:
Are you expecting param to be a specific thing? I really like the idea of piping between these different "apps", but I think it would be nice to document what each expects as an input and what it outputs since unlike most unix CLI tools, this looks like it's expecting something a bit more specified than raw text.

As another side note: I think variable names like `param` in general are somewhat opaque. Would be nice to have a more friendly name so it's a little more clear what type of param is expected since I'm assuming there are some requirements.

Reply:
Agreed with matt on opaque naming. Cool use of `Promise.map` I'm going to start replacing my Promise.all's with this.

Reply:
yes this is my bad, params is a bad name and needs to be documented. I have some thoughts around making this more of a common convention to send data between each app to make it more clear, and also be able to throw meaningful errors when expected data is missing. Also yes typescript would be really useful here to both document and enforce these types.  

I will update this with better comments and naming.